modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManagerImpl.java

```diff
@@ -17,53 +17,282 @@
     public TableManagerImpl(
         ConfigurationManager configurationMgr,
-        ClusterService clusterNetSvc,
         MetaStorageManager metaStorageMgr,
-        SchemaManager schemaMgr,
+        SchemaManager schemaManager,
         Loza raftMgr
     ) {
+        tables = new HashMap<>();
+
         this.configurationMgr = configurationMgr;
-        this.clusterNetSvc = clusterNetSvc;
         this.metaStorageMgr = metaStorageMgr;
-        this.schemaMgr = schemaMgr;
-        this.raftMgr = raftMgr;
+
+        String localNodeName = configurationMgr.configurationRegistry().getConfiguration(NetworkConfiguration.KEY)
+            .name().value();
+
+        configurationMgr.configurationRegistry().getConfiguration(LocalConfiguration.KEY)
+            .metastorageMembers().listen(ctx -> {
+            if (ctx.newValue() != null) {
+                if (hasMetastorageLocally(localNodeName, ctx.newValue()))
+                    subscribeForTableCreation();
+                else
+                    unsubscribeForTableCreation();
+            }
+            return CompletableFuture.completedFuture(null);
+
+        });
+
+        String[] metastorageMembers = configurationMgr.configurationRegistry().getConfiguration(LocalConfiguration.KEY)
+            .metastorageMembers().value();
+
+        if (hasMetastorageLocally(localNodeName, metastorageMembers))
+            subscribeForTableCreation();
+
+        String tableInternalPrefix = INTERNAL_PREFIX + "assignment.#";
+
+        tableCreationSubscriptionFut = metaStorageMgr.registerWatch(new Key(tableInternalPrefix), new WatchListener() {
+            @Override public boolean onUpdate(@NotNull Iterable<WatchEvent> events) {
+                for (WatchEvent evt : events) {
+                    if (!ArrayUtils.empty(evt.newEntry().value())) {
+                        String keyTail = evt.newEntry().key().toString().substring(INTERNAL_PREFIX.length());
+
+                        String placeholderValue = keyTail.substring(0, keyTail.indexOf('.'));
+
+                        UUID tblId = UUID.fromString(placeholderValue);
+
+                        try {
+                            String name = new String(metaStorageMgr.get(
+                                new Key(INTERNAL_PREFIX + tblId.toString())).get()
+                                .value(), StandardCharsets.UTF_8);
+                            int partitions = configurationMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY)
+                                .tables().get(name).partitions().value();
+
+                            List<List<ClusterNode>> assignment = (List<List<ClusterNode>>)IgniteUtils.fromBytes(
+                                evt.newEntry().value());
+
+                            HashMap<Integer, RaftGroupService> partitionMap = new HashMap<>(partitions);
+
+                            for (int p = 0; p < partitions; p++) {
+                                partitionMap.put(p, raftMgr.startRaftGroup(
```

Comment:
Definitely we will need sort of async notifications processing. However it's not the part of current design.